editor: fixes saving point items

jacomyal opened the pull request:
This commit addresses issue #1668 - but does not solve it fully yet.
Point items can properly be saved, but we still have a mismatch between
Turf.js position computation and database positions.

Details:
- Removes geometry from data to save in api.ts
- Stores point item position on track section when dropped
- Refreshes point item coordinates when position is updated from the
left panel form
- Displays a warning when positions from database and Turf.js do not
match

jacomyal committed:
editor: fixes saving point items

This commit addresses issue #1668 - but does not solve it fully yet.
Point items can properly be saved, but we still have a mismatch between
Turf.js position computation and database positions.

Details:
- Removes geometry from data to save in api.ts
- Stores point item position on track section when dropped
- Refreshes point item coordinates when position is updated from the
left panel form
- Displays a warning when positions from database and Turf.js do not
match

sim51 commented:
It works for signals, but by changing the `api.ts` and removing the geometry from the save function, the edition of a track section is broken now.

Moreover, when you select a signal and then click on the `edit` button, the selected point on the map is not red. 
(In fact it is red, but there is a blue point on top)

jacomyal committed:
editor: fixes saving track sections

This commit addresses @sim51 's comment on PR #1733. It injects back the
geometry in track sections properties before saving.
